### PR TITLE
Makefile print better

### PR DESCRIPTION
**Pull request by connorl309**
# Quality Assurance Checklist 
To make reviews more efficient, please make sure the software feature meets the following standards and check everything off that meets the quality check. Once everything has been checked, the assigned reviewers will begin the review process. _Edit this description to check off the list._

There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions. 

## Requirements 
- [x] Followed Coding Style Guide 
- [x] Code Build checks pass 
- [x] No merge conflicts 
- [x] Software feature has test associated with it. 
- [x] Software feature has documentation for it updated.
- [x] If applicable, does software feature have simulation test associated with it (.json & out.json file)
- [x] Test provides useful information and uses relevant data to accurately represent BPS 
    - NOTE: If test file already exists, use that one 
- [x] 2 Members of the team have already reviewed my PR 
    - NOTE: Approver will not look at PR if 2 other members haven't reviewed first 
- [x] Did you Have Fun?

## Things to Consider 
- [ ] Even if the above are checked, is this the best way of writing my code? 
    - It's possible to write code that works, but are there ways to make code more efficient?

### CODE REVIEW

**Comment by manthanand** [comment]
can you post a ss of the output when you make?

**Comment by connorl309** [comment]
> can you post a ss of the output when you make?

image

Defines are specified as "DEFINES=<define name>=VALUE/<define name>=VALUE...."
Forward slash is the separator because Make functions cannot split on commas

**Comment by manthanand** [comment]
I don't think this looks pretty. Also, there shouldn't be errors in the code?

**Comment by manthanand** [comment]
Ig I wasn't too clear in the issue ticket. Could you make it so it prints out the values of all the #defines in the config file every time something is made?

**Comment by connorl309** [comment]
image
There is no possible way in GCC to evaluate our ternary macros, so they will be printed as such.
Works with custom defines and still shows errors if you have invalid value combos such that compilation for other files dependent on the macros fails.

**Comment by manthanand** [comment]
tis a shame. ok

**Comment by manthanand** [comment]
I googled something, could you try this? `XSTR((int)NUM_MINIONS)` ?

**Comment by connorl309** [comment]
> I googled something, could you try this? `XSTR((int)NUM_MINIONS)` for all the defines?

It just prefixes all the prints with "(int)"
image

**Comment by manthanand** [comment]
Its mad stoopid if the ternaries are evaluated in runtime and not compile time.

File: BSP/STM32F413/Makefile [hunk 1]
```diff
@@ -239,14 +250,13 @@ $(BUILD_DIR):
 .INTERMEDIATE: build_msg
 
 c_compile_msg:
-	@echo -e "${GREEN}Compiling C files...${NC}"
```
Comment (Champers5000) [inline BSP/STM32F413/Makefile:242]: echo -e option is needed for escape characters to be interpreted
Why do yall keep getting rid of it ( ˘︹˘ )

**Review by Champers5000 — CHANGES_REQUESTED** [review]

File: BSP/STM32F413/Makefile (hunk 1 above)
Comment (connorl309) [reply BSP/STM32F413/Makefile:242]: What escape characters do you need, the makefile works fine for me with -e being removed. It actually prints the -e on my machine.

Comment (connorl309) [reply BSP/STM32F413/Makefile:242]: image

@Champers5000

Comment (Champers5000) [reply BSP/STM32F413/Makefile:242]: image

That's what happens without the -e on my system
I'll see if there's a way to get it working on both machines

Comment (Champers5000) [reply BSP/STM32F413/Makefile:242]: The escape characters are printed as is in text format and it looks wack af

Comment (Champers5000) [reply BSP/STM32F413/Makefile:242]: @connorl309 What system are you compiling with

Comment (manthanand) [reply BSP/STM32F413/Makefile:242]: this happens for me too which is why I get rid of it

Comment (manthanand) [reply BSP/STM32F413/Makefile:242]: im guessing we are both doing it through wsl

File: Config/Src/config_msgs.c [hunk 2]
```diff
@@ -16,56 +16,90 @@
...
 #define STR(x) #x
 
+#include "config.h"
+
+#ifdef NUM_MINIONS
+#pragma message ("NUM_MINIONS set to " XSTR(NUM_MINIONS))
+#else
+#pragma message ("NUM_MINIONS set to " XSTR(NUM_MINIONS))
+#endif
+
```
Comment (ray4477) [inline Config/Src/config_msgs.c:26]: why is `#pragma message ("NUM_MINIONS set to " XSTR(NUM_MINIONS))`

Comment (ray4477) [reply Config/Src/config_msgs.c:26]: set twice

File: BSP/STM32F413/Makefile (hunk 1 above)
Comment (connorl309) [reply BSP/STM32F413/Makefile:242]: Yes thru WSL

**Comment by connorl309** [comment]
New changes pushed. should be finalized

File: BSP/STM32F413/Makefile [hunk 3]
```diff
@@ -157,15 +157,21 @@ C_INCLUDES =  \
 # compile gcc flags
 ASFLAGS = $(MCU) $(AS_DEFS) $(AS_INCLUDES) $(OPT) -Wall -fdata-sections -ffunction-sections
 
-CFLAGS = $(MCU) $(C_DEFS) $(C_INCLUDES) $(OPT) -Wall -Werror -fdata-sections -ffunction-sections
+CFLAGS = $(MCU) $(C_DEFS) $(C_INCLUDES) $(OPT) -Wall -Werror -fdata-sections -ffunction-sections 
+CFLAGS_SPECIAL = $(CFLAGS)
+CFLAGS_SPECIAL += -ftrack-macro-expansion=0 -fno-diagnostics-show-caret
```
[the remark this answers is not in the document]
Comment (connorl309) [reply BSP/STM32F413/Makefile:162]: let me try, probably but make variables are really finnicky

**Comment by connorl309** [comment]
Ok so tests fail because some fixes from Champer's sim workflow branch aren't in here yet so some compilation explodes bc of array indices

**Review by manthanand — APPROVED** [review]

**Pull request merged**